docs/extensions/builder.py

```diff
@@ -27,6 +27,7 @@ def write_genindex(self) -> None:
         # the total count of lines for each index letter, used to distribute
         # the entries into two columns
         if self.env is None:
```

Comment:
This is where `Builder`s are created and env is not guaranteed to be BuildEnvironment (see typehints). https://github.com/sphinx-doc/sphinx/blob/master/sphinx/registry.py#L157

But in the flow of a Sphinx app this shouldn't be None ever?